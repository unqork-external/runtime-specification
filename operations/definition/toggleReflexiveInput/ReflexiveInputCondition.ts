import { Required, Description } from '@tsed/schema'

import { Unknown } from '../../../../decorators'

export class ReflexiveInputCondition {
  @Required()
  @Description('The component path to the key that should be watched to toggle the reflexive input.')
  pathToValue: string

  @Required()
  @Unknown()
  @Description('The value of `pathToValue` that should toggle the reflexive input.')
  expectedValue: unknown
}
