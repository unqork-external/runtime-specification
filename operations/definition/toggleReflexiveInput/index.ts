export * from './toggleReflexiveInput.model'
export * from './toggleReflexiveInput.options'
