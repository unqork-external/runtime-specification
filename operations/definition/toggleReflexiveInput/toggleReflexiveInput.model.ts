import { DiscriminatorValue, Required } from '@tsed/schema'

import { ToggleReflexiveInputOperationOptions } from './toggleReflexiveInput.options'
import { Alpha, DisplayName } from '../../../../decorators'
import { TrimmedDescription } from '../../../../decorators/schema/trimmedDescription.decorator'
import { AutogenDisabled } from '../../decorators/autogenDisabled.decorator'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DisplayName('Toggle Reflexive Input')
@DiscriminatorValue(OperationTypes.TOGGLE_REFLEXIVE_INPUT)
@TrimmedDescription(`
  Toggles the reflexive input in configuration editor or in ops builder. 
  The reflexive input toggle is determined by whether the target component being watched matches the expected value.
`)
@Alpha()
@AutogenDisabled()
export class ToggleReflexiveInputOperation extends Operation<OperationTypes.TOGGLE_REFLEXIVE_INPUT> {
  @Required()
  public type = OperationTypes.TOGGLE_REFLEXIVE_INPUT as const

  @Required()
  public options: ToggleReflexiveInputOperationOptions = new ToggleReflexiveInputOperationOptions()
}
