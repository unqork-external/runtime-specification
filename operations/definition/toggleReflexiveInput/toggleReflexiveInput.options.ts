import { CollectionOf, Description, MinItems, Required } from '@tsed/schema'

import { ReflexiveInputCondition } from './ReflexiveInputCondition'
import { OperationOptions } from '../../interface'

export class ToggleReflexiveInputOperationOptions extends OperationOptions {
  @Required()
  @Description('The key of the input that should be reflexive.')
  targetKey: string

  @Required()
  @MinItems(1)
  @CollectionOf(ReflexiveInputCondition)
  conditions: ReflexiveInputCondition[]
}
