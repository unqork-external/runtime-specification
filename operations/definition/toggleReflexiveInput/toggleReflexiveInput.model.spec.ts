import { ToggleReflexiveInputOperation } from './toggleReflexiveInput.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: ToggleReflexiveInput Tests', () => {
  it('ToggleReflexiveInput Model', () => {
    const { schema } = generateSchemaAndValidate(ToggleReflexiveInputOperation)

    expect(schema).toMatchSnapshot()
  })
})
