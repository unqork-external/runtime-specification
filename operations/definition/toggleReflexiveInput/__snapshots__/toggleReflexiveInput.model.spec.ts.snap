// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: ToggleReflexiveInput Tests ToggleReflexiveInput Model 1`] = `
{
  "autogenDisabled": true,
  "definitions": {
    "ReflexiveInputCondition": {
      "properties": {
        "expectedValue": {
          "description": "The value of \`pathToValue\` that should toggle the reflexive input.",
          "minLength": 1,
          "type": [
            "string",
            "number",
            "boolean",
            "object",
            "array",
          ],
        },
        "pathToValue": {
          "description": "The component path to the key that should be watched to toggle the reflexive input.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "pathToValue",
        "expectedValue",
      ],
      "type": "object",
    },
    "ToggleReflexiveInputOperationOptions": {
      "properties": {
        "conditions": {
          "items": {
            "$ref": "#/definitions/ReflexiveInputCondition",
          },
          "minItems": 1,
          "type": "array",
        },
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "targetKey": {
          "description": "The key of the input that should be reflexive.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "conditions",
      ],
      "type": "object",
    },
  },
  "description": "Toggles the reflexive input in configuration editor or in ops builder. The reflexive input toggle is determined by whether the target component being watched matches the expected value.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/ToggleReflexiveInputOperationOptions",
    },
    "type": {
      "const": "TOGGLE_REFLEXIVE_INPUT",
      "examples": [
        "TOGGLE_REFLEXIVE_INPUT",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "type": "object",
}
`;
